module/PowerNSX.psm1

```diff
@@ -13947,6 +14024,51 @@ function Set-NsxEdge {
             $_Edge.RemoveChild($edgeSummary) | out-null
         }
 
+        #cliSettings
+        if ( $PsBoundParameters.ContainsKey('userName') ) {
+            if ( $PsBoundParameters.ContainsKey('password') ) {
+                if ( invoke-xpathquery -node $_Edge -querymethod SelectSingleNode -Query "child::cliSettings/userName" ) {
+                    $_Edge.cliSettings.username = $userName
+                } else {
+                    Add-XmlElement -xmlroot $_Edge.cliSettings -xmlElementName "userName" -xmlElementText $userName
+                }
+            } else {
+                throw "You need to specify a password for change username..."
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('password') ) {
+            if ( invoke-xpathquery -node $_Edge -querymethod SelectSingleNode -Query "child::cliSettings/password" ) {
+                $_Edge.cliSettings.password = $password
+            } else {
+                Add-XmlElement -xmlRoot $_Edge.cliSettings -xmlElementName "password" -xmlElementText $password
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('remoteAccess') ) {
+            if ( invoke-xpathquery -node $_Edge -querymethod SelectSingleNode -Query "child::cliSettings/remoteAccess" ) {
+                $_Edge.cliSettings.remoteAccess = $remoteAccess.ToString().ToLower()
+            } else {
+                Add-XmlElement -xmlroot $_Edge.cliSettings -xmlElementName "remoteAccess" -xmlElementText $remoteAccess.ToString().ToLower()
+            }
+        }
+
+        if ( $PsBoundParameters.ContainsKey('passwordExpiry') ) {
+            if ( invoke-xpathquery -node $_Edge -querymethod SelectSingleNode -Query "child::cliSettings/passwordExpiry" ) {
+                $_Edge.cliSettings.passwordExpiry = $passwordExpiry
```

Comment:
This also needs to be cast as a string.

Reply:
Done

Reply:
On what release it don't work ? (because work for me on Windows with PowerShell 6/Core

Reply:
needs to work on Powershell 5.1 on Windows